Lib/_pyrepl/unix_console.py

```diff
@@ -336,7 +336,17 @@ def prepare(self):
         raw.lflag |= termios.ISIG
         raw.cc[termios.VMIN] = 1
         raw.cc[termios.VTIME] = 0
-        tcsetattr(self.input_fd, termios.TCSADRAIN, raw)
+        try:
+            tcsetattr(self.input_fd, termios.TCSADRAIN, raw)
+        except termios.error as e:
+            if e.args[0] == errno.EIO:
+                # gh-135329
+                # When running under external programs (like strace),
+                # tcsetattr may fail with EIO. We can safely ignore this
+                # and continue with default terminal settings.
+                pass
+            else:
+                raise
```

Comment:
This can be simplified:
```suggestion
            if e.args[0] != errno.EIO:
                # gh-135329
                # When running under external programs (like strace),
                # tcsetattr may fail with EIO. We can safely ignore this
                # and continue with default terminal settings.
                raise
```